app/classes/lookup/names.rb

```diff
@@ -60,6 +72,20 @@ def original_matches
     end.flatten.uniq.compact
   end
 
+  def numeric_list?(list)
+    list.all? do |item|
+      case item
+      when Numeric
+        true
+      when String
+        # Handles integers only
+        item.match?(/^-?\d+$/)
```

Comment:
Do you want to match negative numbers? (Nimmo says it's "fed straight to Name.where(id: @vals)")